solr/core/src/java/org/apache/solr/schema/PointField.java

```diff
@@ -158,15 +167,23 @@ public Query getFieldQuery(QParser parser, SchemaField field, String externalVal
       // currently implemented as singleton range
       return getRangeQuery(parser, field, externalVal, externalVal, true, true);
     } else if (field.indexed() && field.hasDocValues()) {
-      Query pointsQuery = getExactQuery(field, externalVal);
+      Query indexQuery = getExactQuery(parser, field, externalVal);
       Query dvQuery = getDocValuesRangeQuery(parser, field, externalVal, externalVal, true, true);
-      return new IndexOrDocValuesQuery(pointsQuery, dvQuery);
+      return new IndexOrDocValuesQuery(indexQuery, dvQuery);
     } else {
-      return getExactQuery(field, externalVal);
+      return getExactQuery(parser, field, externalVal);
     }
   }
 
-  protected abstract Query getExactQuery(SchemaField field, String externalVal);
+  final Query getExactQuery(QParser parser, SchemaField field, String externalVal) {
+    if (hasIndexedTerms(field)) {
+      BytesRefBuilder br = new BytesRefBuilder();
+      readableToIndexed(externalVal, br);
+      return new TermQuery(new Term(field.getName(), br));
+    } else {
+      return getPointRangeQuery(parser, field, externalVal, externalVal, true, true);
```

Comment:
[Q] Prior to this PR, most of the implementing subclasses handled the no-terms-index case by calling Lucene's `IntPoint.newExactQuery(...)`.  Currently, that factory also creates a range-query under the hood, so it's (afaict) functionally equivalent to the `getPointRangeQuery` call here.

If the code's largely equivalent then, is there a reason for the switch?  The Lucene `IntPoint.newExactQuery` call gives us a bit less control I guess, but it has the benefit of being code that we don't have to maintain ourselves.

Are there optimizations in `getPointRangeQuery` that the Lucene equivalent doesn't have?  Or are there other reasons that make the switch worthwhile?